src/stores/DaoStore.ts

```diff
@@ -843,38 +848,54 @@ export default class DaoStore {
       bounty: [],
     };
 
+    // Adds user created proposals that have ended
+    userEvents.newProposal.map(newProposal => {
+      const proposal = this.getProposal(newProposal.proposalId);
+      const votingParameters = this.getVotingParametersOfProposal(
+        newProposal.proposalId
+      );
+
+      if (
+        votingParameters.proposingRepReward.toNumber() > 0 &&
+        isNotActive(proposal)
+      ) {
+        redeemsLeft.rep.push(newProposal.proposalId);
+      }
+    });
+
     // Add possible redeems
     userEvents.votes.map(vote => {
       const proposal = this.getProposal(vote.proposalId);
       const voteParameters = this.getVotingParametersOfProposal(
         vote.proposalId
       );
+
       if (
-        ((proposal.stateInVotingMachine === 1 &&
-          ((proposal.boostedPhaseTime.toNumber() > 0 &&
-            vote.timestamp < proposal.boostedPhaseTime.toNumber()) ||
-            proposal.boostedPhaseTime.toNumber() === 0)) ||
-          (voteParameters.votersReputationLossRatio.toNumber() > 0 &&
-            ((proposal.boostedPhaseTime.toNumber() > 0 &&
-              vote.timestamp < proposal.boostedPhaseTime.toNumber()) ||
-              proposal.boostedPhaseTime.toNumber() === 0) &&
-            proposal.winningVote === vote.vote &&
-            proposal.stateInVotingMachine < 3)) &&
-        redeemsLeft.rep.indexOf(vote.proposalId) < 0
+        (isExpired(proposal) && isBoosted(proposal, vote)) ||

```

Comment:
Awesome that you refactored this, I think we shoudl check it like this, before the proposal has to be expired, and if we have rep locked higher than zero and the vote was done before boosted(if it did boosted, a proposal can execte without reaching boosted) and if the vote was on the winning option
```
isExpired(proposal)
  && (hasLostReputation(voteParameters) && voteBeforeBoosted(proposal, vote))
  && isWinningVote(proposal, vote)
  && redeemsLeft.rep.indexOf(vote.proposalId) < 0)
```